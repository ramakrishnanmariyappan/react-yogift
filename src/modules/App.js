import React, { Component, Suspense, lazy } from "react";
import { Router, Route, Switch, Redirect } from "react-router-dom";
import Header from "./header/components/Header";
import Footer from "./common/components/Footer";
import Landing from "./landing/Landing";
import history from "./common/components/history";
import LinearProgress from "@material-ui/core/LinearProgress";
import { LocalizeProvider } from "react-localize-redux";
import ErrorBoundary from "./common/components/errorBoundary";
import { connect } from "react-redux";

function Loading({ error }) {
  if (error) {
    return (
      <div>
    <h2 style={{
      height: '40px',
      background: '#b3d9f7',
      color: 'white',
      textAlign: 'center',
      verticalAlign: 'middle',
      paddingTop: '5px',
      fontSize: '20px',
      fontWeight: '500'
    }}>
      Oh nooess! Something went wrong. Try re-loading!
      </h2>
      </div>
      )
  } else {
    return <LinearProgress color="secondary" />;
  }
}
export const GiftsListContainer = lazy(() => import('./gifts/containers/GiftsListContainer'));
export const ProfileContainers = lazy(() => import('./user/containers/profileContainers'));
export const GiftShowContainer = lazy(() => import('./gifts/containers/GiftShowContainer'));
export const GiftsSendContainer = lazy(() => import('./user/containers/giftsSendContainer'));
export const GiftsReceivedContainer = lazy(() => import('./user/containers/giftsReceivedContainer'));
export const AddUpdateForm = lazy(() => import('./admin/components/addUpdateForm'));
export const ErrorPage = lazy(() => import('./common/components/ErrorPage'));
export const UsersList = lazy(() => import('./user/components/usersList'));

class App extends Component {
  constructor(props){
    super(props);
    this.state= {}
  }
  render() {
    return (
      <ErrorBoundary>
        <LocalizeProvider>
          <Header />
          <Router history={history}>
            <Suspense fallback={<Loading />}>
              <Switch>
                <Route exact path="/giftCards/:id" render={(props) => <GiftShowContainer {...props} />} />
                <Route exact path="/giftCards" render={() => <GiftsListContainer />} />
                <Route exact path="/Profile" render={() => <ProfileContainers />} />
                <Route exact path="/GiftsSend" render={() => <GiftsSendContainer />} />
                <Route exact path="/UsersList" render={() => <UsersList />} />)
                <Route
                  exact
                  path="/GiftsReceived"
                  render={() => <GiftsReceivedContainer />}
                />
                <Route exact path="/AddUpdateForm" render={() => <AddUpdateForm />} />
                <Route exact path='/AddUpdateForm/:id' render={() => <AddUpdateForm />} />
                <Route exact path="/" component={Landing} />
                <Route exact path="/404" render={() => <ErrorPage />} />
                <Redirect to="/404" />
              </Switch>
            </Suspense>
          </Router>
          <Footer />
        </LocalizeProvider>
       </ErrorBoundary>
    );
  }
}
const mapStateToProps = state => {
  return {
    isLoggedIn: state.login.loginStatus,
    userDetails: state.login.detailsObject
  };
};
export default connect(mapStateToProps,null)(App)