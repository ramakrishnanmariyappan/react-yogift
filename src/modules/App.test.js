import React, { Suspense } from 'react';
import toJson from 'enzyme-to-json';
import App, { GiftsListContainer, ProfileContainers, GiftShowContainer, GiftsSendContainer, GiftsReceivedContainer, AddUpdateForm, ErrorPage, UsersList } from './App';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store'
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure, mount } from 'enzyme';
import { Router } from "react-router-dom";
import history from "./common/components/history";
import { create } from 'react-test-renderer';
import { LocalizeProvider } from "react-localize-redux";
import ErrorBoundary from "./common/components/errorBoundary";
import thunk from 'redux-thunk';

configure({ adapter: new Adapter() });
jest.mock('./Styles.less', () => ({}))
// jest.mock('./LazyComponent1', () => require('./Component1'));
// jest.mock('./LazyComponent2', () => require('./Component2'));

describe('App_Component', () => {
    let store;
    const mockStore = configureStore([thunk]);
    store = mockStore({ login: { loginStatus: true, detailsObject: 'user' } });
    it('rendered lazily', async () => {
        const root = create(
            <Provider store={store}>
                <LocalizeProvider>
                    <Suspense fallback={<div>loading...</div>}>
                        <App />
                    </Suspense>
                </LocalizeProvider>
            </Provider>
        );

        await GiftsListContainer;
        await ProfileContainers;
        await GiftShowContainer;
        await GiftsSendContainer;
        await GiftsReceivedContainer;
        await AddUpdateForm;
        await ErrorPage;
        await UsersList;

        // expect(root).toMatchSnapshot();
    })
});
// describe('Authendication Component', () => {
//     let wrapper, store;
//     beforeEach(() => {
//         const initialState = {
//             isLoggedIn: false,
//             userDetails: 'admin'
//         };
//         const mockStore = configureStore();
//         store = mockStore({ login: initialState });
//         wrapper = shallow(
//             <Provider store={store}>
//                 <Router history={history}>
//                     <App />
//                 </Router>
//             </Provider>
//         );

//     });
//     it('should find the child component', () => {
//         // console.log(wrapper)
//         let instance1 = wrapper.dive().dive().dive().instance();
//     });
// })