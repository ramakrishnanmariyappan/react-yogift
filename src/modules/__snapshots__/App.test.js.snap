// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App_Component rendered lazily 1`] = `
<h1>
  Something went wrong.
</h1>
`;
